Clarify caching

mcecode opened the pull request:
Hi! I'm new to AVA and while I was using it I noticed that subsequent runs didn't seem to finish faster even though I wasn't changing some of my test files. This prompted me to try setting up caching, so I set `"cache": true` in my `package.json` like the docs suggested. However, it didn't seem to make a difference and I couldn't find the specified `node_modules/.cache/ava` directory where the cache is supposed to be. Turns out, [AVA itself doesn't cache files unless used with `@ava/babel`](https://github.com/avajs/ava/issues/2481#issuecomment-622302992). This pull request aims to clarify this, both in the docs and when running `npx ava reset-cache`.

Some notes on the changes I made to the docs:

- I specified that the `cache` option is boolean because some (like I did) might think that it can be set to a directory where the cache will then be set.

Some notes on the changes I made to the `reset-cache` command:

- I removed the return statement at the end of the if check for `resetCache` because it is redundant as either `process.exit()` or `exit()` already terminates the script.
- I changed `console.error()` to `console.log()` because the message logged isn't an error.
- I removed the `nodir: true` option from `del` as it is not a valid option in `fast-glob`, [`fast-glob` instead uses `onlyFiles`](https://github.com/mrmlnc/fast-glob#compatible-with-node-glob) which is then set to `true` by default.

novemberborn commented:
Hi @mcecode, thanks for the PR! Unfortunately I haven't had a chance to look at it yet.

mcecode commented:
It's okay, I know you guys must be busy with the AVA 4 release and other things. So just tell me if I'm missing something or if need to change anything once you get around to review it. Thanks a lot!

novemberborn commented on docs/05-command-line.md:
```diff
@@ -218,13 +218,13 @@ When running a file with and without line numbers, line numbers take precedence.
 ## Resetting AVA's cache
 
-AVA may cache certain files, especially when you use our [`@ava/babel`](https://github.com/avajs/babel) provider. If it seems like your latest changes aren't being picked up by AVA you can reset the cache by running:
+AVA itself does not cache files unless used with our [`@ava/babel`](https://github.com/avajs/babel) provider. However, if it seems like your latest changes aren't being picked up by AVA you can try resetting the cache by running:
```
```suggestion
AVA itself does not cache files unless used with our [`@ava/babel`](https://github.com/avajs/babel) provider. If it seems like your latest changes aren't being picked up by AVA you can try resetting the cache by running:
```

`However` makes it seem like this does apply even if you don't use `@ava/babel`.

novemberborn commented on docs/05-command-line.md:
````diff
@@ -218,13 +218,13 @@ When running a file with and without line numbers, line numbers take precedence.
 ```
 
-This deletes all files in the `node_modules/.cache/ava` directory.
+This notifies you if cache files within the `node_modules/.cache/ava` directory are deleted or if there are no cache files to be deleted.
````
I think the original copy was fine. If it logs that no files were deleted that's OK but that there's no need to mention that here.

mcecode replied:
Agreed, I was trying to be more explicit at the time but I realize now that this extra info is probably unnecessary.

novemberborn commented on docs/06-configuration.md:
```diff
@@ -45,7 +45,7 @@ Arguments passed to the CLI will always take precedence over the CLI options con
 - `ignoredByWatcher`: an array of glob patterns to match files that, even if changed, are ignored by the watcher. See the [watch mode recipe for details](https://github.com/avajs/ava/blob/main/docs/recipes/watch-mode.md)
 - `match`: not typically useful in the `package.json` configuration, but equivalent to [specifying `--match` on the CLI](./05-command-line.md#running-tests-with-matching-titles)
-- `cache`: cache compiled files under `node_modules/.cache/ava`. If `false`, files are cached in a temporary directory instead
+- `cache`: a boolean value on whether to cache compiled files under `node_modules/.cache/ava`. If `false`, files are cached in a temporary directory instead
```
How about this?

```suggestion
- `cache`: defaults to `true` to cache compiled files under `node_modules/.cache/ava`. If `false`, files are cached in a temporary directory instead
```

novemberborn commented on lib/cli.js:
```diff
@@ -244,18 +245,21 @@ export default async () => { // eslint-disable-line complexity
+		}
+
+		try {
+			await del('*', {cwd: cacheDir});
```
Looking at the docs for `del`, it returns the paths of the removed files. So we can check if that's empty and update the messaging based on that. I think that's simpler than having a separate check for the directory.

mcecode replied:
Thanks for pointing that out. I can't believe I missed that 😅, I just defaulted to using `fs`. I've made a commit fixing this.

novemberborn reviewed (commented):
Nice, thanks for this @mcecode! I've left some inline comments. I'll also approve CI now so let's see if that flags anything.

mcecode commented:
Hey @novemberborn, thanks for the feedback! I've added some commits that I think resolve them already. Just looking at the logs, I think the CI failed because the lines I added we're not covered by the tests but I'm not so sure. Just tell me if there's anything else I can do.

novemberborn reviewed (approved):
Awesome, thanks @mcecode!
